script/dump_book.py

```diff
@@ -0,0 +1,399 @@
+    book_data['users'] = list(get_users(set(usernames)))
+
+    book_title = [m['name'] for m in book_data['modules']
+                  if m['portal_type'] == 'Collection'][0]
+    slug = re.sub('[^a-z0-9]+', '-', book_title.lower())
+    if len(slug) <= 5:
+        slug = book_uuid
+    output_filename = 'book_dump.{}@{}.tar'.format(slug, book_version)
+    tmpdir = tempfile.mkdtemp()
+    with tarfile.open(os.path.join(tmpdir, output_filename), 'w') as out:
+        # create one json file per file
+        for i, file_entry in enumerate(book_data.pop('files')):
+            filename = 'files-{:04d}.json'.format(i)
+            with open(os.path.join(tmpdir, filename), 'w') as f:
+                json.dump(file_entry, f)
+            out.add(os.path.join(tmpdir, filename), filename)
+        # create files for other tables
+        for tablename in book_data:
+            filename = '{}.json'.format(tablename)
+            with open(os.path.join(tmpdir, filename), 'w') as f:
+                json.dump(book_data[tablename], f)
+            out.add(os.path.join(tmpdir, filename), filename)
+    if os.path.exists(output_filename):
+        os.remove(output_filename)
+    shutil.move(os.path.join(tmpdir, output_filename), '.')
+    shutil.rmtree(tmpdir)
+    print('Output in {}'.format(output_filename))
+
+
+def load_table(cursor, tablename, columns, data):
+    sql = 'INSERT INTO {} ({}) VALUES ({})'.format(
+        tablename, ', '.join(columns), ', '.join('%s' for c in columns))
+    for d in data:
+        with db_cursor(cursor_factory=None) as cursor:
+            try:
+                cursor.execute(sql, d)
+            except psycopg2.errors.UniqueViolation as e:
+                logging.error(e)
+
+
+def load_data(tablename, data):
+    print('loading data into {}'.format(tablename))
+    if not data:
+        return
+    columns = list(data[0].keys())
+    with db_cursor(cursor_factory=None) as cursor:
+        load_table(cursor, tablename, columns,
+                   (tuple(d[c] for c in columns) for d in data))
+
+
+def confirm_load():
+    module_count = execute_sql('SELECT count(*) FROM modules')[0]['count']
+    if module_count == 0:
+        return True
+    hostname = socket.gethostname()
+    sys.stdout.write('The modules table is not empty. You are on {}. '
+                     'Confirm load. (yes/No) '.format(hostname))
+    if sys.version_info.major == 2:
+        # raw_input is not defined in python 3
+        confirmation = raw_input()  # noqa
+    else:
+        confirmation = input()
+    return confirmation.lower() == 'yes'
+
+
+def load_book(filename):
+    infile = tarfile.open(filename, 'r')
+
+    def get_data(field):
+        return json.loads(infile.extractfile('{}.json'.format(field)).read()
+                          .decode('utf-8'))
+
+    # Disable all the triggers except ...
+    with db_cursor() as cursor:
+        cursor.execute('ALTER TABLE modules DISABLE TRIGGER ALL')
```

Comment:
```
python dump_book.py load_book book_dump.tar 
Running: load_book book_dump.tar
Traceback (most recent call last):
  File "dump_book.py", line 399, in <module>
    globals()[sys.argv[1]](arg)
  File "dump_book.py", line 328, in load_book
    cursor.execute('ALTER TABLE modules DISABLE TRIGGER ALL')
  File "/usr/local/lib/python2.7/dist-packages/psycopg2/extras.py", line 243, in execute
    return super(RealDictCursor, self).execute(query, vars)
psycopg2.errors.InsufficientPrivilege: permission denied: "RI_ConstraintTrigger_a_281330" is a system trigger
```

I ended up needing to use a super user to load the book. Not an issue AFAICT, just making note.

Reply:
Yeah, that's not too bad.  Loading is only for development machines / databases anyway, so shouldn't be a problem :)